src/propIs.js

```diff
@@ -0,0 +1,28 @@
+var _curry3 = require('./internal/_curry3');
+var is = require('./is');
+var propSatisfies = require('./propSatisfies');
+
+
+/**
+ * Returns `true` if the specified object property is of the given type;
+ * `false` otherwise.
+ *
+ * @func
+ * @memberOf R
+ * @category Type
+ * @sig Type -> String -> Object -> Boolean
+ * @param {Function} type
+ * @param {String} name
+ * @param {*} obj
+ * @return {Boolean}
+ * @see R.is
+ * @see R.propSatisfies
+ * @example
+ *
+ *      R.propIs('x', Number, {x: 1, y: 2});  //=> true
+ *      R.propIs('x', Number, {x: 'foo'});    //=> false
+ *      R.propIs('x', Number, {});            //=> false
+ */
+module.exports = _curry3(function propIs(type, name, obj) {
+  return propSatisfies(is(type), name, obj);
+});
```

Comment:
I don't consider the behaviour depicted in the diagrams above complex. In fact, this is how I assumed `pipe` to work when I first considered applying it to functions of more than one argument.

`useWith` is a solution in this case but it sidesteps the question of how `compose` and `pipe` should handle functions of more than one argument.